lib/cfilters.c

```diff
@@ -709,3 +703,50 @@ void Curl_pollset_change(struct Curl_easy *data,
   }
 }
 
+static void ps_add(struct Curl_easy *data, struct easy_pollset *ps,
+                   int bitmap, curl_socket_t *socks)
+{
+  if(bitmap) {
+    int i;
+    for(i = 0; i < MAX_SOCKSPEREASYHANDLE; ++i) {
+      if(!(bitmap & GETSOCK_MASK_RW(i)) || !VALID_SOCK((socks[i]))) {
+        break;
+      }
+      if(bitmap & GETSOCK_READSOCK(i)) {
+        if(bitmap & GETSOCK_WRITESOCK(i))
+          Curl_pollset_add_inout(data, ps, socks[i]);
+        else
+          Curl_pollset_add_in(data, ps, socks[i]);
+      }
+      else
```

Comment:
Add a comment like /* must be bitmap & GETSOCK_WRITESOCK(i) */ to show the implication.

Reply:
Ack.